Add additionalProperties to vertex ai Schema definition

Eitan1112 opened the pull request:
Add additionalProperties field to vertex ai Schema TypedDict

## Title

Add additionalProperties to vertex ai Schema definition

## Relevant issues

Fixed #14251

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [ ] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

I tested the fix locally and it's definitely fixing the bug, but I don't have any more time to invest into this issue and PR to understand the tests structure and contribute a test to this very specific issue.

## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🐛 Bug Fix

## Changes
Add additionalProperties to vertex ai Schema definition

Eitan1112 committed:
Add additionalProperties to vertex ai Schema definition

Add additionalProperties field to vertex ai Schema TypedDict

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=14252) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=14252) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=14252) it.</sub>

Eitan1112 committed:
Change additionalProperties type to Any

This is aligned with "default" which is also `Any`, and both in vertex ai docs:
https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.cachedContents#Schema

are both with 'value' type
